import_export/mixins.py

```diff
@@ -8,51 +8,78 @@
 from .signals import post_export
 
 
-class ExportViewMixin:
+class BaseImportExportMixin:
     formats = base_formats.DEFAULT_FORMATS
-    form_class = ExportForm
     resource_class = None
 
+    def get_resource_class(self):
+        if not self.resource_class:
+            return modelresource_factory(self.model)
+        return self.resource_class
+
+    def get_resource_kwargs(self, request, *args, **kwargs):
+        return {}
+
+
+class BaseImportMixin(BaseImportExportMixin):
+    def get_import_resource_class(self):
+        """
+        Returns ResourceClass to use for import.
+        """
+        return super().get_resource_class()
+
+    def get_import_formats(self):
+        """
+        Returns available import formats.
+        """
+        return [f for f in self.formats if f().can_import()]
+
+    def get_import_resource_kwargs(self, request, *args, **kwargs):
+        return super().get_resource_kwargs(request, *args, **kwargs)
+
+
+class BaseExportMixin(BaseImportExportMixin):
+    model = None
+
     def get_export_formats(self):
         """
         Returns available export formats.
         """
         return [f for f in self.formats if f().can_export()]
 
-    def get_resource_class(self):
-        if not self.resource_class:
-            return modelresource_factory(self.model)
-        return self.resource_class
-
     def get_export_resource_class(self):
         """
         Returns ResourceClass to use for export.
         """
-        return self.get_resource_class()
-
-    def get_resource_kwargs(self, request, *args, **kwargs):
-        return {}
+        return super().get_resource_class()
 
     def get_export_resource_kwargs(self, request, *args, **kwargs):
-        return self.get_resource_kwargs(request, *args, **kwargs)
+        return super().get_resource_kwargs(request, *args, **kwargs)
 
-    def get_export_data(self, file_format, queryset, *args, **kwargs):
-        """
-        Returns file_format representation for given queryset.
-        """
+    def get_data_for_export(self, request, queryset, *args, **kwargs):
         resource_class = self.get_export_resource_class()
-        data = resource_class(**self.get_export_resource_kwargs(self.request))\
+        return resource_class(**self.get_export_resource_kwargs(request, *args, **kwargs))\
             .export(queryset, *args, **kwargs)
-        export_data = file_format.export_data(data)
-        return export_data
 
-    def get_export_filename(self, file_format):
+    def get_filename(self, file_format):
```

Comment:
I'd say that you can keep this function as `get_export_filename` and don't implement it in `ExportViewMixin`, but not important.

Reply:
I might have misunderstood you, but I think this runs into the problem of `get_export_filename()` being called in two places with the different args.

Reply:
Only the `admin.py` uses the different arguments, and you are already *adapting* that call inside `ExportMixin`:

https://github.com/django-import-export/django-import-export/pull/1277/files#diff-a6dabb2139e7ec78138476e9a12311c22cc93cbeab4a93994d4f95f64934805eR437-R438

But I could be wrong :)